### PR TITLE
Pagination

### PR DESCRIPTION
**Pull request by wrnkld**
This pull request touches a few different pages, adds images and moves other images around.

The content views have minor changes (with references that don’t yet go to patternfly.org but instead reference github).

The pagination section under navigation has significant updates to reflect JIRA PTNFLY-2090.

### CODE REVIEW

**Comment by wrnkld** [comment]
@mcarrano @serenamarie125

File: pattern-library/navigation/pagination/design/overview.md [hunk 1]
```diff
@@ -9,6 +9,7 @@ This pattern defines how pagination behaves within content views. When present,
   * Pagination is not necessary when there is a small amount of content.
   * Pagination is not necessary when an application uses "lazy load" to load content as a user scrolls.
 * Pagination can be "sticky" and remain fixed to the bottom of a user's browser.
```
Comment (mcarrano) [inline pattern-library/navigation/pagination/design/overview.md]: Thanks for providing the added clarification @wrnkld .  I'd also like @andresgalante to review this statement as we were discussing the idea of fixed footers at the browser level and whether we want that.  Alternatively the pagination would be sticky to the bottom of the table or list container.  Andres, please comment.

**Comment by wrnkld** [comment]
@jgiardino @andresgalante @LHinson @mcarrano 
I've made a few more updates to the pagination design.
1. Added more callouts and explanation to the original design
2. Adding a section called **Pagination - Content Below**
3. Added a section called **Selection While Paging**

File: pattern-library/navigation/pagination/design/overview.md [hunk 2]
```diff
@@ -1 +1,16 @@
-Words here
+# Pagination
+
+## Overview
```
Comment (LHinson) [inline pattern-library/navigation/pagination/design/overview.md]: remove this header to be consistent with other pattern pages layout/format

File: pattern-library/navigation/pagination/design/overview.md [hunk 3]
```diff
@@ -1 +1,16 @@
...
+
+This pattern defines how pagination behaves within content views. When present, pagination is consistent across all three content views.
+
+* Pagination is found below the content.
```
Comment (LHinson) [inline pattern-library/navigation/pagination/design/overview.md]: Not sure what you mean by "is found". Do you mean that pagination is located at the bottom of the content view?

File: pattern-library/navigation/pagination/design/overview.md [hunk 4]
```diff
@@ -1 +1,15 @@
...
+* Pagination is optional.
+  * Pagination is not necessary when there is a small amount of content.
+  * Pagination is not necessary when an application uses "lazy load" to load content as a user scrolls.
+* Pagination can be "sticky" and remain fixed to the bottom of a user's browser.
```
Comment (LHinson) [inline pattern-library/navigation/pagination/design/overview.md]: After our conversation, we have agreed that this is out of scope for this story and additional stories will be added to enhance this pattern page. I just need the context to add stories to the backlog. This comment is not blocking approval for this particular PR.

File: pattern-library/navigation/pagination/design/design.md [hunk 5]
```diff
@@ -1,3 +1,56 @@
-## Pagination Design
+# Pagination
 
-Words here
+## Pagination Elements
```
Comment (LHinson) [inline pattern-library/navigation/pagination/design/design.md]: Please remove this header

File: pattern-library/navigation/pagination/design/design.md [hunk 6]
```diff
@@ -1,3 +1,56 @@
...
+
+![Pagination Callouts](img/pagination_callouts.png)
+
+1. Number of items on a page
```
Comment (LHinson) [inline pattern-library/navigation/pagination/design/design.md]: Labels should be bold and have header style capitalization. I would try and name them something more succinct (e.g. "Items per Page" and then the description text can get in more detail so... "Items per Page: This control identified the how many items are being listed and allows a user to change the number of items per page."

File: pattern-library/navigation/pagination/design/design.md [hunk 7]
```diff
@@ -1,3 +1,56 @@
...
+  * Includes option to modify the number of items displayed. It is recommended that the default number be less than 25 in order to minimize scrolling.
```
Comment (LHinson) [inline pattern-library/navigation/pagination/design/design.md]: See comment above. Should address rest of the list as well.

File: pattern-library/navigation/pagination/design/design.md [hunk 8]
```diff
@@ -1,3 +1,56 @@
...
+
+## Pagination - Selection Pattern
+
+### Default View
+![Pagination Select 1](img/pagination_selection_1.png)
+
+### Filters Applied
+![Pagination Select 2](img/pagination_selection_2.png)
+
+1. Filters displayed on left side.
```
Comment (LHinson) [inline pattern-library/navigation/pagination/design/design.md]: All list items should have a label.

**Comment by LHinson** [comment]
@serenamarie125 @maryclarke @catrobson please review. Thanks!

File: pattern-library/navigation/pagination/design/design.md [hunk 9]
```diff
@@ -1,3 +1,56 @@
...
+![Pagination Select 4](img/pagination_selection_4.png)
+
+1. Select all items within a single page by checking the box in the column header.
+
+### Select All
+![Pagination Select 5](img/pagination_selection_5.png)
+
+1. Select all items within the table by clicking Select All link.
+
+### Clear All
```
Comment (serenamarie125) [inline pattern-library/navigation/pagination/design/design.md]: Does anyone think that "Clear Selections" may be better here?  When I looked at the mock, i found it a bit confusing that there are both "Clear All Filters" and "Clear All".  Since "Clear All" isn't near the checkboxes, I'm wondering if changing the label may also help with identifying what's being cleared.  Thoughts @mcarrano @LHinson @maryclarke @wrnkld ?

File: pattern-library/navigation/pagination/design/overview.md [hunk 10]
```diff
@@ -1 +1,16 @@
...
+  * A sticky footer is not recommended for views with content *below* pagination.
```
Comment (serenamarie125) [inline pattern-library/navigation/pagination/design/overview.md]: Does it make sense for us to say that a sticky footer is optional, but recommended for all cases except for views with content below pagination ?
@wrnkld @LHinson @maryclarke @catrobson @mcarrano

File: pattern-library/navigation/pagination/design/design.md (hunk 9 above)
Comment (mcarrano) [inline pattern-library/navigation/pagination/design/design.md]: Yeah, maybe.  This did not bother me the first time I looked at it, but I see your point @serenamarie125.  This also raises a slightly different question.  Looks like I only get Clear All after I've chosen to Select All.  But isn't it valid to let the user clear all selections when less than all items have been selected?  Should Select All and Clear All be present together?

File: pattern-library/navigation/pagination/design/overview.md (hunk 4 above)
Comment (mcarrano) [inline pattern-library/navigation/pagination/design/overview.md]: I think that part of the problem is it's not clear what we mean by "sticky footer."  We don't actually have such a thing in PatternFly today as I understand it.  In discussing with @andresgalante it seemed like making this sticky to the bottom of the browser was not the right answer.  Perhaps this just confuses the matter and the statement should be removed for now?

**Comment by LHinson** [comment]
@wrnkld @serenamarie125 I've added story to address the following in the next sprint. 
* No data to view 
* no data matching filters 

See: https://patternfly.atlassian.net/browse/PTNFLY-2161

**Comment by wrnkld** [comment]
I made the formatting updates. But there are still a few issues mentioned above not addressed in the current design...

- Should sticky footer be referenced or not?
- Should Clear All (or Clear Selected) be present at all times?
- A recommendation for no data to view.
- A recommendation for no data matching filters.

**Comment by LHinson** [comment]
Thanks @wrnkld for making those updates and identifying the additional items to address. FYI @serenamarie125 @mcarrano that we have included these additional items in Matthew's story this sprint. 

I think we can go ahead and merge this one and wait for another PR for Matthew to update. Any concerns?

**Comment by serenamarie125** [comment]
Sounds good @LHinson, thanks for capturing these issue in the follow on story.  Nice work @wrnkld , looking forward to seeing the next iteration too!
